lib/elliptic/ec/signature.js

```diff
@@ -23,35 +23,74 @@ function Signature(options, enc) {
 }
 module.exports = Signature;
 
+function getLength(buf, p) {
+  var initial = buf[p.place++];
+  if (!(initial & 0x80)) {
+    return initial;
+  }
+  var octetLen = initial & 0xf;
+  var byteLength = octetLen;
+  var val = buf[p.place + --byteLength];
+  var mul = 1;
+  while (byteLength > 0 && (mul *= 0x100)) {
+    val += buf[p.place + --byteLength] * mul;
+  }
+  p.place += octetLen;
+  return val;
+}
 Signature.prototype._importDER = function _importDER(data, enc) {
   data = utils.toArray(data, enc);
-  if (data.length < 6 || data[0] !== 0x30 || data[2] !== 0x02)
-    return false;
-  var total = data[1];
-  if (1 + total > data.length)
-    return false;
-  var rlen = data[3];
-  // Short length notation
-  if (rlen >= 0x80)
-    return false;
-  if (4 + rlen + 2 >= data.length)
+  var p = {};
```

Comment:
Let's do a `function Position() { this.place = 0; }` for this, should help compiler to optimize this.